src/training/gen_patches.py

```diff
@@ -0,0 +1,58 @@
+import random
+import numpy as np
+
+def get_rand_patch(img, mask, sz=160):
+    """
+    :param img: ndarray with shape (x_sz, y_sz, num_channels)
+    :param mask: binary ndarray with shape (x_sz, y_sz, num_classes)
+    :param sz: size of random patch
+    :return: patch with shape (sz, sz, num_channels)
+    """
+    assert len(img.shape) == 3 and img.shape[0] > sz and img.shape[1] > sz and img.shape[0:2] == mask.shape[0:2]
```

Comment:
This can be changed to an `if` statement instead of an assert and return `None, None`

```suggestion
    if not (len(img.shape) == 3 and \
            img.shape[0] > sz and \
            img.shape[1] > sz and \
            img.shape[0:2] == mask.shape[0:2]):
            return None, None
```
This will test the same condition. If your condition is True then it will continue with the rest of the code. If not, it will return `None, None`.

Having said that, if you need it to raise an error instead of return we can raise an error. Generally `assert` usage is not recommended in functions and methods. It's kind of tricky on how it works.